tests/test_configuration.py

```diff
@@ -450,6 +452,42 @@ def test_guess_from_coordinate_mapping_no_sym(self):
 
         assert eq.is_nested()
 
+    # pytest -s -k test_hairy_ball_theorem tests/test_configuration.py
+    @pytest.mark.unit
+    def test_hairy_ball_theorem(self):
+        """Ensure |B| does not vanish after coordinate mapping."""
+        eq = get("W7-X")
+        rho = np.linspace(0, 1, 5)
+        alpha = np.linspace(0, 2 * np.pi, 2)
+        zeta = np.linspace(0, 4 * np.pi, 10)
+
+        grid_jit, data_jit = _field_line_to_desc_coords(
+            eq, rho, alpha, zeta, jitable=True
+        )
+        data_jit = eq.compute("|B|", grid=grid_jit, data=data_jit)
+        grid_no_jit, data_no_jit = _field_line_to_desc_coords(
+            eq, rho, alpha, zeta, jitable=False
+        )
+        data_no_jit = eq.compute("|B|", grid=grid_no_jit, data=data_no_jit)
+        print(data_jit["|B|"])
+        print()
+        print(data_no_jit["|B|"])
+        # I think there are two separate bugs.
+        # Bug 1:
+        # Regardless of any logic issues with the coordinate mapping etc., I think
```

Comment:
When `jitable=True` we don't scale the weights to account for duplicates, that could be causing some issue? Though if those weights are being used somewhere for a surface average that likely indicates something else is wrong?

Reply:
I don't think it has to do with duplicate nodes. I commented my debugging results below